src/sage/schemes/generic/morphism.py

```diff
@@ -669,6 +710,201 @@ def glue_along_domains(self, other):
+            sage: R.<x> = QQ[]
+            sage: J = HyperellipticCurve(x^5 - 8*x).jacobian()
+            sage: phi = End(J).identity(); phi
+            Scheme endomorphism of Jacobian of Hyperelliptic Curve over Rational Field defined by y^2 = x^5 - 8*x
+              Defn: Identity map
+            sage: psi = SchemeMorphism_sum([phi, phi, phi]); psi
+            Sum morphism:
+              From: Jacobian of Hyperelliptic Curve over Rational Field defined by y^2 = x^5 - 8*x
+              To:   Jacobian of Hyperelliptic Curve over Rational Field defined by y^2 = x^5 - 8*x
+              Via:  (Scheme endomorphism of Jacobian of Hyperelliptic Curve over Rational Field defined by y^2 = x^5 - 8*x
+              Defn: Identity map, Scheme endomorphism of Jacobian of Hyperelliptic Curve over Rational Field defined by y^2 = x^5 - 8*x
+              Defn: Identity map, Scheme endomorphism of Jacobian of Hyperelliptic Curve over Rational Field defined by y^2 = x^5 - 8*x
+              Defn: Identity map)
+            sage: phi + phi == phi + phi
+            True
+            sage: psi == phi + phi + phi
+            True
+
+            sage: P = J(2, 4); (P, psi(P), P * 3)
+            ((2, y - 4), (1), (1))
+
+        The codomain should implement addition::
+
+            sage: phi = Spec(QQ).identity_morphism()
+            sage: SchemeMorphism_sum([phi, phi])
+            Traceback (most recent call last):
+            ...
+            ValueError: addition is not implemented for Spectrum of Rational Field
+
+        The zero morphism can be defined when the codomain implements addition::
+
+            sage: SchemeMorphism_sum([], Spec(QQ), ZZ)
+            Sum morphism:
+              From: Spectrum of Rational Field
+              To:   Spectrum of Integer Ring
+              Via:  ()
+        """
+        phis = tuple(phis)
+
+        if not phis and (domain is None or codomain is None):
+            raise ValueError('need either phis or both domain and codomain')
+
+        for phi in phis:
+            if not isinstance(phi, SchemeMorphism):
+                raise ValueError(f'not a scheme morphism: {phi}')
+
+        if domain is None:
+            domain = phis[0].domain()
+        if codomain is None:
+            codomain = phis[0].codomain()
+        if check:
+            try:
+                x = codomain.an_element()
+                _ = x + x
+            except (TypeError, NotImplementedError):
+                raise ValueError(f"addition is not implemented for {codomain}")
+        for phi in phis:
+            if phi.domain() != domain:
+                raise ValueError(f'summand {phi} has incorrect domain (need {domain})')
+            if phi.codomain() != codomain:
+                raise ValueError(f'summand {phi} has incorrect codomain (need {codomain})')
+
+        self._phis = phis
+        self._domain = domain
+        SchemeMorphism.__init__(self, self._domain, codomain)
```

Comment:
This line is wrong... The second argument is the `parent` (homset) not the domain 😓